File: 1833-find-the-highest-altitude/1833-find-the-highest-altitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        int[] altitudes = new int[gain.length + 1];
        altitudes[0] = 0;
        int max = 0;
        for(int i = 0; i < gain.length; i++){
            altitudes[i+1] = altitudes[i] + gain[i];
            max = Math.max(max, altitudes[i+1]);
        }
        return max;
    }
}

File: 1833-find-the-highest-altitude/1833-find-the-highest-altitude.rs
impl Solution {
    pub fn largest_altitude(gain: Vec<i32>) -> i32 {
        let mut maxi = 0;
        let mut curr_altitude = 0;
        for g in gain{
            curr_altitude += g;
            maxi = maxi.max(curr_altitude);
        }
        maxi
    }
}